### ISSUE TITLE
All caps buttons and hyperlinks - not WCAG compliant?

### ISSUE THREAD

**Issue by sigmatec** [Dogfalo/materialize#4745]
Not so much an issue as a topic for discussion...

I realise that Google's Material Design states that buttons should be displayed in all caps, so this framework is compliant with that.  However, WCAG recommends not using all caps in any circumstances.  There are only **five** instances of `text-transform: uppercase` in the main CSS file - should these be omitted by default?

I am having to un-text-transform these in my application because some of our clients are public sector bodies (UK) and they are anally retentive about WCAG compliance.

**Comment by DanielRuf** [comment]
Makes much sense imho https://github.com/humanmade/hm-pattern-library/issues/75

Can you link the WCAG (2.0) guideline / rule (is it 3.1)?

**Comment by Dogfalo** [comment]
I would argue that caps on one word isn't going to severely impact readability. This is certainly a problem if you start having sentences in caps. Also I have confidence that what's good enough for Google in terms of accessibility is going to be good enough for most people. If they change this later in the design spec we will too.

**Comment by sigmatec** [comment]
@Dogfalo I agree completely with you - my personal preference is that the ALLCAPS in the buttons is actually easier to read.

@DanielRuf It used to be 3.1 of WCAG 2.0 [https://www.w3.org/WAI/WCAG20/quickref/?showtechniques=31#meaning](url) however I've just had a good look through and can't actually find any reference to over-use of capitalisation anymore. Hopefully it's gone and I can tell our clients that it's no longer relevant next time they mention it!

**Comment by DanielRuf** [comment]
> @DanielRuf It used to be 3.1 of WCAG 2.0 however I've just had a good look through and can't actually find any reference to over-use of capitalisation anymore. Hopefully it's gone and I can tell our clients that it's no longer relevant next time they mention it!

Exactly. I just saw
> Using upper and lower case according to the spelling rules of the text language (future link)

But it seems they do not get much clearer and also in the WCAG C22 technique rules they use an uppercase class for .
https://www.w3.org/TR/WCAG20-TECHS/C22.html#C22-ex7

So it looks like this is ok now?

**Comment by sigmatec** [comment]
As far as I can see, yes.  Whether or not our clients we see it in the same way remains to be seen!

**Comment by dotjay** [comment]
Just stumbled across this thread and thought I'd throw my 2p worth in…

This issue is primarily discussed as part of [Techniques for the The Cognitive and Learning Disabilities Accessibility Task Force (COGA)](https://w3c.github.io/coga/techniques/). Due to the impact that ALLCAPS can have on different groups of users, the issue can be considered to fall into WCAG 2's Level AAA success criteria, 3.1.5 Reading Level and 3.1.6 Pronunciation.

The general advice is to avoid writing text in ALLCAPS for a number of reasons:
1. Screen readers may read each letter out individually
  - Consider the difference between "Contact us" and "CONTACT US". When the text is all in capital letters, software is unable to analyse the text to know whether it is more appropriate to say "contact us" or "contact U.S." so it is common for people to hear the less appropriate text announced by their screen reader.
  - Interestingly, some screen readers treat "Add to cart" differently to "ADD TO CART", announcing the latter as "A D D to cart".
2. People don't like listening to shouting
  - Using ALLCAPS is considered by some to be the digital equivalent of shouting. In fact, some screen readers announce uppercase letters with extra emphasis, which can be a little like listening to shouting.
3. ALLCAPS are harder for some people to read
  - Capital letters are harder for people to read ([13–18% harder, say Gov.UK](https://www.gov.uk/guidance/content-design/writing-for-gov-uk#capital-letters-are-harder-to-read)). It is [disputed as to why capital letters are harder to read](https://www.blog.theteamw.com/2009/12/23/100-things-you-should-know-about-people-19-its-a-myth-that-all-capital-letters-are-inherently-harder-to-read/), and how much it impacts readability, but it is best avoided.
